strax/storage/s3.py

```diff
@@ -78,44 +93,33 @@ def _find(self, key, write, fuzzy_for, fuzzy_for_options):
 
         Search the S3 store to see if data is there.
         """
-        # Don't support fuzzy matching since hard without RunDB
         if fuzzy_for or fuzzy_for_options:
             raise NotImplementedError("Can't do fuzzy with S3")
 
         # Check exact match / write case
         key_str = str(key)
         bk = self.backend_key(key_str)
 
-        # Get list of all buckets / (run_id, lineage)
-        objects_list = self.s3client.list_buckets()
-
-        # If no buckets exist...
-        if 'Buckets' not in objects_list:
-            # No objects yet... so no metadata
+        # See if any objects exist for this key
+        objects_list = self.s3.list_objects(Bucket=BUCKET_NAME,
```

Comment:
Nice! So we don't list the entire universe anymore.

Reply:
Yup